plugins/cursecheck.cpp

```diff
@@ -160,8 +160,9 @@ command_result cursecheck (color_ostream &out, vector <string> & parameters)
     {
         df::unit * unit = world->units.all[i];
 
+        // filter out all "living" units that are currently removed from play
         // don't spam all completely dead creatures if not explicitly wanted
-        if(unit->flags1.bits.dead && ignoreDead)
+        if((unit->flags1.bits.dead && !unit->flags2.bits.killed) || (unit->flags2.bits.killed && ignoreDead))
```

Comment:
It would be clearer to use library functions from #1297 here once that's merged.

Reply:
True, but I can't do that before then, at least not if I wan't it to compile in my end.